llvm/include/llvm/ADT/APInt.h

```diff
@@ -2193,6 +2193,12 @@ inline const APInt absdiff(const APInt &A, const APInt &B) {
   return A.uge(B) ? (A - B) : (B - A);
 }
 
+/// Compute the higher order bits of unsigned multiplication of two APInts
+APInt mulhu(const APInt &C1, const APInt &C2);
```

Comment:
Could you clarify by what 'higher order bits' mean? Mathematically, is it `(C1 * C2) % Max(C2)`?

Reply:
no, it is first C2 bits afte the MSB of (c1 * c2) , where (c1 * c2) is of (C2 * 2) bits.

Reply:
Ah sorry, what I wrote is the definition of low-order bits. I meant: `(C1 * C2) % Max(C2)` or `(C1 * C2) >> bitwidth(C2)`. My point is that I'd appreciate if the documentation explained what's being calculated, and having the exact formula would help.

Reply:
Sure thing.